tetris.py

```diff
@@ -125,23 +125,55 @@ def new_block(self):
             self.current_block_shape = self.upcoming_block_shape
         self.current_block_center = (int(game_width / 2), -2)
         self.upcoming_block_shape = random.choice(shape_names)
+        self.index = 0
 
     def get_current_block(self):
         (x, y) = self.current_block_center
         if self.current_block_shape == "I":
-            return [(x - 2, y), (x - 1, y), (x, y), (x + 1, y)]
+            I = [
+                [(x - 2, y), (x - 1, y), (x, y), (x + 1, y)],
+                [(x, y - 2), (x, y - 1), (x, y), (x, y + 1)],
+            ]
+            return I[self.index % len(I)]
         if self.current_block_shape == "L":
-            return [(x - 1, y - 1), (x - 1, y), (x, y), (x + 1, y)]
+            L = [
+                [(x - 1, y), (x, y), (x + 1, y), (x + 1, y - 1)],
+                [(x + 1, y), (x, y), (x, y - 1), (x, y - 2)],
+                [(x, y), (x, y - 1), (x + 1, y - 1), (x + 2, y - 1)],
+                [(x, y - 1), (x + 1, y - 1), (x + 1, y), (x + 1, y + 1)],
+            ]
+            return L[self.index % len(L)]
         if self.current_block_shape == "L_rev":
-            return [(x - 1, y), (x, y), (x + 1, y), (x + 1, y + 1)]
+            L_rev = [
+                [(x - 1, y - 1), (x, y - 1), (x + 1, y - 1), (x + 1, y)],
+                [(x, y), (x + 1, y), (x + 1, y - 1), (x + 1, y - 2)],
+                [(x, y - 1), (x, y), (x + 1, y), (x + 2, y)],
+                [(x + 1, y - 1), (x, y - 1), (x, y), (x, y + 1)],
+            ]
+            return L_rev[self.index % len(L_rev)]
         if self.current_block_shape == "O":
-            return [(x - 1, y), (x, y), (x + 1, y), (x + 1, y + 1)]
+            O = [[(x - 1, y), (x, y), (x, y - 1), (x - 1, y - 1)]]
+            return O[0]
         if self.current_block_shape == "E":
-            return [(x - 1, y), (x, y), (x + 1, y), (x + 1, y + 1)]
+            E = [
+                [(x - 1, y), (x, y), (x + 1, y), (x, y - 1)],
+                [(x, y - 1), (x, y), (x + 1, y), (x, y + 1)],
+                [(x - 1, y), (x, y), (x + 1, y), (x, y + 1)],
+                [(x, y), (x - 1, y), (x, y - 1), (x, y + 1)],
+            ]
+            return E[self.index % len(E)]
         if self.current_block_shape == "Z":
-            return [(x - 1, y), (x, y), (x + 1, y), (x + 1, y + 1)]
+            Z = [
+                [(x - 1, y - 1), (x, y - 1), (x, y), (x + 1, y)],
+                [(x, y), (x, y - 1), (x + 1, y - 1), (x + 1, y - 2)],
+            ]
+            return Z[self.index % len(Z)]
         if self.current_block_shape == "Z_rev":
-            return [(x - 1, y), (x, y), (x + 1, y), (x + 1, y + 1)]
+            Z_rev = [
+                [(x + 1, y - 1), (x, y - 1), (x, y), (x - 1, y)],
+                [(x, y), (x, y - 1), (x - 1, y - 1), (x - 1, y - 2)],
+            ]
+            return Z_rev[self.index % len(Z_rev)]
```

Comment:
Cleanup idea: Every part of this function ends with `return some_variable[self.index % len(some_variable)]`. If you use the same variable name for every variable, let's say `coord_lists`, it becomes `return coord_lists[self.index % len(coord_lists)]`, and then because the exact same code is used in each case, it can be written only once, at the end of the function.